libraries/AP_HAL_SITL/Semaphores.cpp

```diff
@@ -76,4 +76,77 @@ bool Semaphore::take_nonblocking()
     return false;
 }
 
+
+/*
+  binary semaphore using pthread condition variables
+ */
+
+BinarySemaphore::BinarySemaphore(bool initial_state) :
+    AP_HAL::BinarySemaphore(initial_state)
+{
+    pthread_cond_init(&cond, NULL);
```

Comment:
same point as raised in SITL implementation. The reasoning behind the choice of pthread_cond_* api over sem_* api to implement this.

Reply:
- sem_ doesn't work on MacOSX
- you can't avoid a race in signal that leads to count going over 1 without adding another mutex